a10_octavia/common/utils.py

```diff
@@ -191,15 +191,21 @@ def convert_interface_to_data_model(interface_obj):
     interface_dm = data_models.Interface()
     if not interface_num:
         raise exceptions.MissingInterfaceNumConfigError()
+    if not isinstance(interface_num, int):
```

Comment:
In all the below cases, it's highly unlikely that these are inheriting from some Integer superclass. Therefore, `type()` is a more appropriate check

Reply:
ok